### DISCUSSION TITLE
Why need Edge in the clustering?

### DISCUSSION THREAD

**Discussion by plawansai** [OvenMediaLabs/OvenMediaEngine#1220] [Help]
Could I implement the clustering without Edge?
https://airensoft.gitbook.io/ovenmediaengine/origin-edge-clustering

Thank you very much.

**Comment by bchah** [comment]
You would need to build your own Edge or master load balancer that could handle communication with the pool of Origin servers and routing / requesting streams for the users. While implementing such a solution will let you have exact control over routing of the streams / server capacity, there are many advantages to just using Origin-Edge, including:

- Allows scaling of viewers without duplicating streams across multiple servers.
- Allows better geographic distribution of streams. E.g. Origin in Canada with Edges in Europe + Asia will provide a better playback experience to the remote viewers.
- Far less risk that an Origin server would become overwhelmed since it only has to receive the original streams and pass to the Edge servers.
- Finally, Origin-Edge handles much of the logic you would need to create to address an Origin-Only pool.

I should also mention that Airensoft offers some very helpful technical support plans for OME: https://airensoft.com/support.html - if you are using Origin-Edge then they can help you with support but if you create your own solution it may not be as easy for them to help.
